haskellPackages: update stackage and hackage

expipiplus1 opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/Mw5WLnzsP7fC4Zky) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #154461. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

piyush-kurur committed:
haskellPackages.raaz: disable parallel building

Parallel builds seems to interfere with backpack type checking.
See: https://gitlab.haskell.org/ghc/ghc/-/issues/17188 for more details.

ulysses4ever committed:
haskellPackages.hs-speedscope: unbreak

expipiplus1 committed:
haskellPackages: stackage-lts 18.21 -> 18.22

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

expipiplus1 committed:
all-cabal-hashes: 2022-01-14T12:47:41Z -> 2022-01-18T22:54:05Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

expipiplus1 committed:
Merge pull request #155520 from ulysses4ever/hs-speedscope-jailbreak

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.nix-tree: reflect brick 0.65 -> 0.66 update

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

expipiplus1 committed:
haskellPackages.hnix: pin at < 0.15

ulysses4ever committed:
haskellPackages.eventlog2html: unbreak

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.hnix: drop obsolete patch

sternenseemann commented:
Last notable regression is [arbtt](https://hydra.nixos.org/build/164948454) cc @rvl.

sternenseemann committed:
haskell.lib: move lib.nix into same directory as compose.nix

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

expipiplus1 committed:
Merge pull request #156206 from sternenseemann/haskell-lib-move

haskell.lib: move lib.nix into same directory as compose.nix

cdepillabout committed:
arbtt: jailbreak because of tasty-golden version constraint

cdepillabout commented:
> Last notable regression is arbtt

I sent a PR for this: https://github.com/NixOS/nixpkgs/pull/156407

(I'm also an arbtt user)

expipiplus1 committed:
Merge pull request #156407 from cdepillabout/arbtt

expipiplus1 committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

expipiplus1 committed:
haskellPackages.gi-adwaita: disable on darwin

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh
